prow/gerrit/adapter/adapter.go

```diff
@@ -166,12 +174,39 @@ func (c *Controller) applyGlobalConfigOnce(cfg config.Getter, gerritClient *clie
 	gerritClient.Authenticate(cookiefilePath, tokenPathOverride)
 }
 
+// Helper function to create the cache used for InRepoConfig. Currently only attempts to create cache and returns nil if failed.
+func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, configAgent *config.Agent) (cache *config.InRepoConfigCache, err error) {
+	opts := git.ClientFactoryOpts{
+		CloneURI:       cloneURI.String(),
+		Host:           cloneURI.Host,
+		CookieFilePath: cookieFilePath,
+	}
+	gc, err := git.NewClientFactory(opts.Apply)
+	if err != nil {
+		//TODO(mpherman): Return Error once we know this WAI
+		logrus.Warn("Failed to create Gerrit Client for InRepoConfig")
+		return nil, nil
+	}
+	// Initialize cache for fetching Presubmit and Postsubmit information. If
+	// the cache cannot be initialized, exit with an error.
+	cache, err = config.NewInRepoConfigCache(
+		cacheSize,
+		configAgent,
+		config.NewInRepoConfigGitCache(gc))
+	// If we cannot initialize the cache, exit with an error.
+	if err != nil {
+		logrus.WithField("in-repo-config-cache-size", cacheSize).WithError(err).Fatal("unable to initialize in-repo-config-cache")
+		return nil, nil
+	}
+	return cache, nil
+}
+
 // Sync looks for newly made gerrit changes
 // and creates prowjobs according to specs
 func (c *Controller) Sync() error {
 	syncTime := c.tracker.Current()
 	latest := syncTime.DeepCopy()
-
+	repoCacheMap := map[string]*config.InRepoConfigCache{}
```

Comment:
I guess each repo is assigned its own cache (even though the InRepoConfigCache is already keyed by the repo identifier string (so it can handle multiple repos)), because each repo needs its own cookeFilePath for auth? If so, I would mention this in the comments here.

Wait, I'm still confused why this is necessary because the cache itself doesn't care about auth (it's not an argument the cache takes when we use it). 🤔

Reply:
I think it's because of `cloneURI` is different among each host, it's required for cloning repo

Reply:
`Sync` is called every minute from upstream, which means this `repoCacheMap` is re-initialized every minute. Can we make this part of `Controller`, so that the cache persist among each run?

Reply:
Done